xclim/subset.py

```diff
@@ -434,16 +464,16 @@ def create_mask(
 def subset_shape(
     ds: Union[xarray.DataArray, xarray.Dataset],
     shape: Union[str, Path, gpd.GeoDataFrame],
+    vectorize: bool = False,
```

Comment:
After time tests I think we would want to make vectorize the default.   In fact is there any reason to keep the sjoin() version at all?   We might want to test performance a bit more I suppose

Reply:
Anyways my small number of tests are pretty conclusive that vectorize is faster so I would put it as the dafault